src/script/lua_api/l_util.cpp

```diff
@@ -320,6 +321,103 @@ int ModApiUtil::l_decompress(lua_State *L)
 	return 1;
 }
 
+#if USE_CURL
+void ModApiUtil::read_http_fetch_request(lua_State *L, HTTPFetchRequest &req)
+{
+	luaL_checktype(L, 1, LUA_TTABLE);
+
+	req.caller = httpfetch_caller_alloc();
+	getstringfield(L, 1, "url", req.url);
+	req.post_data = getstringfield_default(L, 1, "post_data", "");
+	std::string useragent = getstringfield_default(L, 1, "useragent", "");
+	lua_getfield(L, 1, "useragent");
+	bool useragent_exists = lua_isstring(L, 2);
+	lua_pop(L, 1);
+	if (useragent_exists)
+		req.useragent = getstringfield_default(L, 1, "useragent", "");
```

Comment:
Oh thanks, I should have taken a closer look at that code. I just pushed
updated code that changes that function according to your suggestion.
I'm not really familiar with the whole lua <--> C API, so my code was
primarily based on what other scriptapi functions used.